client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -103,24 +104,146 @@ export const siteSetupFlow: Flow = {
 		const verticalsStepEnabled = isEnabled( 'signup/site-vertical-step' ) && isEnabledFTM;
 		const goalsStepEnabled = isEnabled( 'signup/goals-step' ) && isEnabledFTM;
 
-		// Set up Step progress for Woo flow - "Step 2 of 4"
-		if ( intent === 'sell' && storeType === 'power' ) {
-			switch ( currentStep ) {
-				case 'storeAddress':
-					setStepProgress( { progress: 1, count: 4 } );
-					break;
-				case 'businessInfo':
-					setStepProgress( { progress: 2, count: 4 } );
-					break;
-				case 'wooConfirm':
-					setStepProgress( { progress: 3, count: 4 } );
-					break;
-				case 'processing':
-					setStepProgress( { progress: 4, count: 4 } );
-					break;
+		switch ( currentStep ) {
```

Comment:
Could you make a hook for this part? Maybe something like `useProgressBar` or `useUpdateFlowProgress` and include a README (better as it's public) to describe how to update it or link to your p2 post. Then we can keep `useStepNavigation` function more simple.

Reply:
Agree

Reply:
@arthur791004, I updated the PR for `useUpdateFlowProgress` update, can we do a review for the latest changes? Thanks so much!!